### PR TITLE
Code didn't quite work as expected - added tests

### PR DESCRIPTION
**Pull request by lloydwatkin**
I've also added a `.travis.yml` file if you'd like to hook that up :)

@jacquesdev added the inverse behaviour 

To clarify: your code did what you expected, but it wasn't quite right as it would call the callback multiple times - which also meant our inverse functionality couldn't work. I've prevented this from happening.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,11 +19,26 @@ Grep through all mocha tests for any `only` tests.
...
+        .pipe(grepFail('123', {inverse: true}));
+    });
+
 ## API
 
-`grepFail(predicates)`
+`grepFail(predicates, [options])`
 
   * **predicates** `(string|array)` string(s) to grep for and fail if they exist
+  * **options.inverse** `(boolean) - optional` default behaviour is to throw an error if a match is found. Optionally, you can set this param to true so that it fails if it is expecting values to match.
```
Comment (noseglid) [inline README.md]: Maybe we could make this clearer:

`default behavior is to throw an error if a match is found. Set`options.inverse`to`true`to throw an error if the match is *not* found.`

File: index.js [hunk 2]
```diff
@@ -8,34 +8,52 @@ var PluginError = gutil.PluginError;
...
 var PLUGIN_NAME = 'gulp-grep-fail';
 
-function grepFailPlugin (predicates) {
+function grepFailPlugin (predicates, options) {
   if (!predicates) {
     throw new PluginError(PLUGIN_NAME, 'No predicates specified.');
   }
+  if (!options) {
+    options = {};
+  }
```
Comment (noseglid) [inline index.js]: This is normally written as `options = options || {}`.

**Comment by noseglid** [comment]
Awesome with the tests! I have some minor comments, but I'd be happy to merge this!

**Comment by lloydwatkin** [comment]
Updated :)

**Comment by noseglid** [comment]
Cool! Thanks for this!

**Pull request merged**